File: KClosest.java
//Time complexity: O log(n+k)
//Space complexity: O (1)

class Solution {
    public List<Integer> findClosestElements(int[] arr, int k, int x) {

        int low = 0;
        int high = arr.length - k;

        while (low < high) {
            int mid = low + (high - low) / 2;

            int startD = x - arr[mid];
            int endD = arr[mid + k] - x;
            if (startD > endD)
                low = mid + 1;
            else
                high = mid;
        }

        List<Integer> res = new ArrayList<>();
        for (int i = low; i < low + k; i++)
            res.add(arr[i]);

        return res;
    }
}

File: k-closest.py
# TC: O(log(N-K)) :  
# SC: O(1) 
# Did this code successfully run on Leetcode : Yes
# Any problem you faced while coding this : No

class Solution:
    def findClosestElements(self, arr: List[int], k: int, x: int) -> List[int]:
        if len(arr)==0:
            return []
        
        low = 0
        high = len(arr)-1
        
        while(high-low+1>k):
            distL = x-arr[low]
            distH = arr[high]-x
            if distL>distH:
                low+=1
            else:
                high-=1
                
        result = arr[low:high+1]
        return result